Orange/preprocess/score.py

```diff
@@ -207,7 +207,7 @@ def _gini(D):
     """Gini index of class-distribution matrix"""
     P = D / np.sum(D, axis=0)
     return sum((np.ones(1 if len(D.shape) == 1 else D.shape[1]) - np.sum(np.square(P), axis=0))
-               * 0.5 * np.sum(D, axis=0) / np.sum(D))
+               * np.sum(D, axis=0) / np.sum(D))
```

Comment:
At first glance it looks to me like at least the second line is needed to get the weighted impurity we want.
It is possible that 1 can be used instead of the complications with np.ones (should be tested).